src/test/ui/impl-trait/incoherent-assoc-imp-trait.rs

```diff
@@ -0,0 +1,16 @@
+// Regression test for issue 67856
```

Comment:
Let's move this to https://github.com/varkor/rust/tree/master/src/test/ui/type-alias-impl-trait.